### ISSUE TITLE
[Bug]: puppeteer.connect is hanging if the alert popup is shown during connection

### ISSUE THREAD

**Issue by umeshaccion** [puppeteer/puppeteer#9729]
### Bug expectation

I am using Lighthouse + puppeteer in node JS code to connect **already opened browser**, run lighthouse in snapshot mode and generate reports. Everything is working fine except one scenario where I have to run this node JS  when there is default alert popup opened in browser. Node js not executing fully, it is getting stuck after the code where we connect with browser (with puppeteer.connect method). after that I have printed lot of logs but I don't see those logs.

Please let me know if any questions.

Steps to reproduce:

1) Launch Chrome with the remote debugging port, e.g., `/Applications/Google\ Chrome\ Canary.app/Contents/MacOS/Google\ Chrome\ Canary --remote-debugging-port=9222 --user-data-dir=/tmp/pptr-connect`
2) Open DevTools and execute `alert('test')` in the DevTools console.
3) Run the script below.

### Bug behavior

- [ ] Flaky
- [ ] PDF

### Minimal, reproducible example

```TypeScript
const puppeteer = require('puppeteer');

(async function () {
  try {
    const browser = await puppeteer.connect({
      browserURL: 'http://127.0.0.1:9222',
    });
    console.log('connect')
    const pages = await browser.pages();
    console.log('after pages', pages);
  } catch (err) {
    console.error(err);
  }
})()
```

### Error string

NO error

### Puppeteer version

19.7.2

### Node version

16.16.0

### Package manager

npm

### Package manager version

7.5.0

### Operating system

Windows

**Comment by OrKoN** [comment]
Is the browser running in headless or headful mode? Have you tried handling the popup explicitly?

```
page.on('dialog', (dialog) => {
      console.log(dialog.message());
      dialog.accept(); 
})
```

**Comment by OrKoN** [comment]
Also, is the browser version compatible with the puppeteer version you are using? https://pptr.dev/chromium-support

**Comment by umeshaccion** [comment]
@OrKoN  
control is not reaching at this point in code when alert popup is there in browser and running browser in headful mode. 

page.on('dialog', (dialog) => {
      console.log(dialog.message());
      dialog.accept(); 
});


Browser Version is : Version 110.0.5481.97 (Official Build) (64-bit)

Tried below both the versions.

puppeteer version : 19.7.2 
puppeteer version : 19.4.0

**Comment by OrKoN** [comment]
And it does only reproduce with Lighthouse?

**Comment by OrKoN** [comment]
I think I am able to reproduce. Lighthouse does not look related at all here. I will update the steps to reproduce.

**Comment by umeshaccion** [comment]
Yes as you said, Lighthouse doesn't matter here.

**Comment by OrKoN** [comment]
It looks like the browser stops responding to some CDP commands when the alert is shown. We either need to work around this on Puppeteer side or on the browser side.

**Comment by OrKoN** [comment]
Also, as soon as the alert dialog is closed manually connection establishes without a problem.

**Comment by umeshaccion** [comment]
> Also, as soon as the alert dialog is closed manually connection establishes without a problem.

Yes that I observed. when you click OK on popup it will again establish the connection. but the thing is when we are running some automation, we will be stuck whenever we will get these types popups.

Thank you for your response.

**Comment by OrKoN** [comment]
@umeshaccion the workaround would be to connect before any site is loaded. What is your use case for connecting after the site is loaded?

**Comment by umeshaccion** [comment]
> @umeshaccion the workaround would be to connect before any site is loaded. What is your use case for connecting after the site is loaded?

Unfortunately this workaround will not work for us, use case I have mentioned above.

**Comment by wgorski** [comment]
We have the same problem. Also, if there are multiple tabs in the browser, a dialog in ANY of the open tabs will block puppeteer from connecting to the browser.
It would be nice if puppeteer could connect despite the fact that a dialog is open.
